Source/WebCore/rendering/RenderListBox.cpp

```diff
@@ -805,13 +805,13 @@ static void setupWheelEventTestMonitor(RenderListBox& renderer)
 
 void RenderListBox::setScrollTop(int newTop, const ScrollPositionChangeOptions&)
 {
-    // Determine an index and scroll to it.    
+    // Determine an index and scroll to it.
     int index = newTop / itemHeight();
-    if (index < 0 || index >= numItems() || index == m_indexOffset)
+    if (index == m_indexOffset)
         return;
 
     setupWheelEventTestMonitor(*this);
-    scrollToOffsetWithoutAnimation(ScrollbarOrientation::Vertical, index);
+    scrollToOffsetWithoutAnimation(ScrollbarOrientation::Vertical, std::clamp(index, 0, numItems() - 1));
```

Comment:
I think we should continue to return early (and not call either of these two functions) if after clamping `index == m_indexOffset`.

Also, will this do the right thing if `numItems() == 0`? `std::clamp(0, 0, -1) == -1`. Although maybe we'll have returned early above in this case?

Reply:
Hmm yeah good catch. I'll fix this.